torch/csrc/autograd/profiler_python.cpp

```diff
@@ -38,11 +38,18 @@ static constexpr size_t CallTypeSize = 3;
 struct CodeLocation {
   CodeLocation() = default;
   explicit CodeLocation(const PyFrameObject* frame)
+  // clang-format off
+#if PY_VERSION_HEX >= 0x030B0000
```

Comment:
Could we have a python compat header file that defines the macro as suggested in https://docs.python.org/3.11/whatsnew/3.11.html?
In particular, for this, we can use as suggested
```c
#if PY_VERSION_HEX < 0x030900B1
static inline PyCodeObject* PyFrame_GetCode(PyFrameObject *frame)
{
    Py_INCREF(frame->f_code);
    return frame->f_code;
}
#endif
```

That way, there is no need to have ifdef/clang-format disable everywhere in the codebase.

Reply:
> I would decline your idea to maintain compat declarations.

Ho interesting. Why do you think that having this functions defined once in a header and re-used in all the places we need is not as good as the specific ifdef that are currently in this PR?

Reply:
> > I would decline your idea to maintain compat declarations.
> 
> Ho interesting. Why do you think that having this functions defined once in a header and re-used in all the places we need is not as good as the specific ifdef that are currently in this PR?

Due to lack of time, I stop author here.
Feel free fork out for quick move over it.

Reply:
Ok, I was wondering if there were technical reasons.
I can take over this if you don't have time for the updates of course.

Thanks a lot for taking the time to make the first draft!